docs/operational_guide/upgrading_m3.md

```diff
@@ -0,0 +1,73 @@
+# Upgrading M3
+
+## Overview
+
+This guide explains how to upgrade (and downgrade??) M3 from one version to another (e.g. from 0.14.0 to 0.15.0) - both on Kubernetes as well as on bare metal.
```

Comment:
@robskillington is it always unconditionally safe to downgrade M3DB version?

If not, we should mention downgrading should only be done if the release notes say it's safe.

Reply:
Yeah, wasn't sure what to put here, but that sounds reasonable.